data_processor.py

```diff
@@ -1,60 +1,55 @@
 import os,sys,csv
 
 class DataProcessor(object):
-    '''The constructor of the DataProcessor'''
-    def __init__(self, input_file):
-        self.input_file = input_file
-
-    '''load_file method loads the file from current folder and check if it's valid , with open will close file if any excetption happed '''
-    def load_file(self):
-        with open(self.input_file) as f:
-            data=f.readlines()
-            for line in data:
-                print(line)
+
+    '''load_file method loads the file from the current folder and check if it's valid , with open will close file if any
+    excetption happend '''
+    def load_file(self, input_file):
+        with open(input_file) as csv_file:
+            data=csv_file.readlines()
             return data
 
     '''filter each person and get the highest salary record for each person.
         By default , the input records are sorted.'''
-    def filterInfo(self, data):
-        resultDict={}
-        firstline = True
-        for line in data[:-1]:
-            if firstline:
-                firstline = False
-                continue
-            
+    def filter_info(self, data):
+        result_dict={}
+
+        LEGAL_LINE_LENGTH = 7
+        INDEX_OF_SALARY = 5
+        INDEX_OF_EMPLOYEE_ID = 1
+
+        '''Skip the header of the file.'''
+        for line in data[1:]:            
             words = line.split(",")
-            print(words[5])
+
+            '''If the current line is not long enough then skip'''
+            if len(words) != LEGAL_LINE_LENGTH:
+                continue 
             
-            if words[1] not in resultDict:
-            	resultDict[str(words[1])]=line
+            if words[INDEX_OF_EMPLOYEE_ID] not in result_dict:
+                result_dict[words[INDEX_OF_EMPLOYEE_ID]] = line
             else:
-                print("test value "+words[1])
-                print(resultDict.get(''+words[1]))
-                oldWords = resultDict.get(words[1]).split(",")
+                old_words = result_dict.get(words[INDEX_OF_EMPLOYEE_ID]).split(",")
                 
-                if int(words[5])> int(oldWords[5]):
-                    resultDict[''+str(words[1])]=line
-            print("Dict: ")
-            print(resultDict)
+                if int(words[INDEX_OF_SALARY])> int(old_words[INDEX_OF_SALARY]):
```

Comment:
we do not know what actually will be in `words[INDEX_OF_SALARY]` or if it can convert into int.
will suggest have a helper function like 
```
def isfloat(value):
  try:
    float(value)
    return True
  except ValueError:
    return False```